apps/srp.c

```diff
@@ -553,6 +553,8 @@ int srp_main(int argc, char **argv)
             user = *(argv++);
         else {
             user = NULL;
+            if (mode == OPT_LIST)
+                break;
```

Comment:
As for logic. I don't know much about apps/srp.c, but if it possible that there are limitations on -list usage? I mean is it possible that you shouldn't be able to mix -list with some other options? What I'm trying to say is that logic might have to be refined... Once again, I don't actually know apps/srp.c... Bottom line is that question to *you* is if it makes sense to do something differently and simpler than rather mechanically follow what was there.